app/serializers/current_user_serializer.rb

```diff
@@ -342,4 +344,12 @@ def include_grouped_unread_notifications?
   def include_unseen_reviewable_count?
     redesigned_user_menu_enabled
   end
+
+  def redesigned_user_page_nav_enabled
+    if SiteSetting.enable_new_user_profile_nav_groups.present?
+      GroupUser.exists?(user_id: object.id, group_id: SiteSetting.enable_new_user_profile_nav_groups.split("|"))
```

Comment:
Instead of doing another query you could just rely on `groups` here, and see if there are any for the user that match the IDs in `SiteSetting.enable_new_user_profile_nav_groups.split("|")` .

Side note, I'm trying to add some shortcuts to these group ID checks in https://github.com/discourse/discourse/pull/18042 since we are going to be using them more often, hopefully I can merge that soon after the security fixes go out.

Reply:
I thought about that but I can't rely on `groups` because `groups` are restricted to just visible groups of the user.

Reply:
Hmm that's awkward, I feel like we should just load all the `GroupUser` records in this serializer then filter appropriately to avoid these double ups, but that's not specific to just this PR

Reply:
I thought about that but since this feature flag is meant to be short lived, the overhead of an additional cheap query is fine IMO.